### DISCUSSION TITLE
Change date formats to universal

### DISCUSSION THREAD

**Discussion by DrJohnM61** [teslamate-org/teslamate#676] [Ideas]
**What would you like to be added**:
US format dates are used and these are confusing to UK users as we use DD/MM/YY. Recommend using non ambiguous format DD-MMM-YY

**Why is this needed**:
To avoid confusion for global users

**Additional context**:

**Comment by Koofteh123** [comment]
Select to edit table, go to second tab and there you can change the date format.

24 hour clock makes my brain hurt, I was so glad when I finally switched it to AM/PM. 😁

**Reply by stauffenberg2020** [reply]
Where is this second tab link in Grafana 8.5. Can you please print a screenshot

**Comment by DrJohnM61** [comment]
@Koofteh123 Thank you. I can update the view or even the JSON, however, won't that get overwritten if there is a new release? Not a grafana expert...

![image](https://user-images.githubusercontent.com/59447364/80952163-2247ff80-8df1-11ea-8e54-f562763c37ee.png)

also, if I update the view, I have to save it as a local copy:
Cannot save provisioned dashboard
This dashboard cannot be saved from Grafana's UI since it has been provisioned from another source. Copy the JSON or save it to a file below. Then you can update your dashboard in corresponding provisioning source.
See documentation for more information about provisioning.

**Comment by Koofteh123** [comment]
Yes, it will get overwritten with an update. I learned this the hard way. 😕

You have to copy the JSON code when it tells you it can't save a provisioned dashboard. Then click the gear icon on top of the page, go to JSON model tab and paste in the code you copied.

**Comment by DrJohnM61** [comment]
> Yes, it will get overwritten with an update. I learned this the hard way. 😕
> 
> You have to copy the JSON code when it tells you it can't save a provisioned dashboard. Then click the gear icon on top of the page, go to JSON model tab and paste in the code you copied.

Thanks - I appreciate the advice and this is something that I am aware of and have experimented with. My request was for an enhancement so that this could be a universal change to the base code. My suggested format looks that it will work without taking much reporting space (IMHO), would be globally acceptable as a date format and be persistent if applied to the base JSON code.

**Comment by Str1atum** [comment]
What about changing the date/time format in some global configuration file of TeslaMate - is there already a way to adjust all formatting for Grafana?

**Comment by stauffenberg2020** [comment]
Hi All, Is anyone really successful in changing the current date format to non ambiguous formats (like DD-MMM-YYYY hh:mm:ss a) in (all) the dashboards which displays the dates? If so, I would be happy to see a screenshot of where the change needs to be done.

**Comment by duncanbarnes** [comment]
Ideally configurable on install would be great to see but if a choice has to be made surely the most universal and unambiguous is ISO 8601

**Comment by stauffenberg2020** [comment]
Hi, Is it still being looked upon or has this idea/request been forgotten?

**Comment by NelisV** [comment]
Would love to see this too! 
Has @DrMichael's work on this led to something?
_bump_

**Comment by DrMichael** [comment]
I would say, that is obsolete here, because we use now the browser locale (and date format) in Grafana.
https://github.com/teslamate-org/teslamate/pull/4064

**Reply by Pretagonist** [reply]
This is still an issue. I have all my software in English but I don't use the date and time formats that they do in the UK or the US. 

Language is not a date format

**Reply by swiffer** [reply]
true - as far as i know browsers are not sending there prefered date_formats - that's why grafana opted into relying on the locale assuming 99% of the users are not changing their locale from os defaults. and that functionality is what we rely on.

https://grafana.com/docs/grafana/latest/setup-grafana/configure-grafana/#use_browser_locale

**Reply by stauffenberg2020** [reply]
> 99% of the users are not changing their locale from os defaults

I am not sure if this a right fact

**Reply by swiffer** [reply]
true - and within users of teslamate the percentage might be even higher.
but maybe most of these are fine with having date_format changing as well or are even expecting this.

any proposal on what can be done instead?
